src/UI/Component/Modal/Modal.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\JavaScriptBindable;
+use ILIAS\UI\Component\Trigger\Triggerable;
+use ILIAS\UI\Component\Trigger\TriggerAction;
+
+/**
+ * This describes commonalities between the different modals
+ */
+interface Modal extends Component, JavaScriptBindable, Triggerable
+{
+
+    /**
+     * Get the title of the modal
+     *
+     * @return string
+     */
+    public function getTitle();
+
+
+    /**
+     * Get the component representing the content of the modal
+     *
+     * @return \ILIAS\UI\Component\Component
+     */
+    public function getContent();
+
+
+    /**
+     * Get all buttons of the modal
+     *
+     * @return \ILIAS\UI\Component\Button\Button[]
+     */
+    public function getButtons();
```

Comment:
@alex40724 You would inject this button into the footer of the modal, in the case of a RoundTrip modal it would look like this:
```php
$button = $factory->ui()->button()->standard('Download All', 'your-action');
$modal = $factory->ui()->modal()->roundtrip('My Title', $myContent)->withButtons([$button]);
```